packages/scene-composer/src/utils/entityModelUtils/modelShaderComponent.ts

```diff
@@ -34,8 +34,22 @@ export const createModelShaderEntityComponent = (shader: IColorOverlayComponent)
   return comp;
 };
 
-export const updateModelShaderEntityComponent = (modelShader: IColorOverlayComponent): ComponentUpdateRequest => {
+export const updateModelShaderEntityComponent = (
+  modelShader: IColorOverlayComponent,
+  oldComponent?: IColorOverlayComponent,
+): ComponentUpdateRequest => {
   const request = createModelShaderEntityComponent(modelShader);
+  if (oldComponent) {
```

Comment:
`if` check not needed. Covered by `(oldComponent.ruleBasedMapId || oldComponent.valueDataBinding)`.